packages/flutter/test/material/app_bar_test.dart

```diff
@@ -2339,6 +2341,32 @@ void main() {
       expect(tester.getSize(findAppBarMaterial()).height, kToolbarHeight);
     });
 
+    testWidgets('backgroundColor animation', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        buildAppBar(contentHeight: 1200.0, surfaceTintColor: Colors.transparent),
+      );
+
+      expect(getAppBarAnimatedBackgroundColor(tester), defaultColor);
+
+      TestGesture gesture = await tester.startGesture(const Offset(50.0, 400.0));
+      await gesture.moveBy(const Offset(0.0, -kToolbarHeight));
+      await gesture.up();
+      await tester.pump();
+
+      expect(getAppBarAnimatedBackgroundColor(tester), defaultColor);
+      await tester.pumpAndSettle();
+      expect(getAppBarAnimatedBackgroundColor(tester), scrolledColor);
```

Comment:
I don't think this test case verifies the feature. It only verifies the starting color and ending color, while you're trying to verify that there's a smooth animation in between. I recommend going without `pumpAndSettle` but `pump`ing a time instead, and it should result in a intermediate color.

Reply:
I have pushed changes for 2 intermediate color.
1. At 50ms after animation start
2. At 100ms after animation start

Animation duration is 200ms.